src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -123,18 +121,7 @@ public class FailoverConnectionPlugin implements IConnectionPlugin {
   private final PropertySet propertySet;
   private final IConnectionPlugin nextPlugin;
   private final Log logger;
-  private final Pattern auroraDnsPattern =
-      Pattern.compile(
-          "(.+)\\.(proxy-|cluster-|cluster-ro-|cluster-custom-)?([a-zA-Z0-9]+\\.[a-zA-Z0-9\\-]+\\.rds\\.amazonaws\\.com)",
-          Pattern.CASE_INSENSITIVE);
-  private final Pattern auroraCustomClusterPattern =
-      Pattern.compile(
-          "(.+)\\.(cluster-custom-[a-zA-Z0-9]+\\.[a-zA-Z0-9\\-]+\\.rds\\.amazonaws\\.com)",
-          Pattern.CASE_INSENSITIVE);
-  private final Pattern auroraProxyDnsPattern =
-      Pattern.compile(
-          "(.+)\\.(proxy-[a-zA-Z0-9]+\\.[a-zA-Z0-9\\-]+\\.rds\\.amazonaws\\.com)",
-          Pattern.CASE_INSENSITIVE);
+  private final RdsUtils rdsHelper;
```

Comment:
do you need an instance of that class or do you just need to call the methods this class contains?